packages/terra-avatar/src/Avatar.jsx

```diff
@@ -43,56 +44,92 @@ const defaultProps = {
+  static generateImage(image, variant, alt) {
+    const avatarIconClassNames = cx([
+      'avatar-icon',
+      variant,
+    ]);
+
+    const icon = <span className={avatarIconClassNames} />;
+
+    return <TerraImage className={cx('avatar-image')} src={image} placeholder={icon} alt={alt} />;
   }
 
-  const attributes = Object.assign({}, customProps);
+  constructor(props) {
+    super(props);
 
-  const avatarClassNames = cx([
-    'avatar',
-    attributes.className,
-  ]);
+    // If image has been provided we need to generate the image to display and store it in the state
+    if (props.image) {
+      const { alt, image, variant } = props;
 
-  const avatarIconClassNames = cx([
-    'avatar-icon',
-    variant,
-  ]);
+      const imageComponent = Avatar.generateImage(image, variant, alt);
+      this.state = { imageComponent };
+    }
 
-  const avatarTextClassNames = cx([
-    { 'avatar-text-small': initials && initials.length === 3 },
-    { 'avatar-text-large': initials && initials.length === 2 },
-  ]);
+    // Checks to run when not in production
+    if (process.env.NODE_ENV !== 'production') {
+      if (props.image && props.initials) {
+        // eslint-disable-next-line
+        console.warn('Only one of the props: [image, initials] should be supplied.');
+      }
+    }
+  }
 
-  const icon = <span className={avatarIconClassNames} />;
+  componentWillReceiveProps(newProps) {
+    // If we have an image to display (they take precedence) and one of its attributes have changed
+    if (newProps.image && (newProps.image !== this.props.image ||
+      newProps.alt !== this.props.alt || newProps.variant !== this.props.variant)) {
+      const { alt, image, variant } = newProps;
 
-  let avatarContent;
-  if (image) {
-    avatarContent = <TerraImage className={cx('avatar-image')} src={image} placeholder={icon} alt={alt} />;
-  } else if (initials && (initials.length === 2 || initials.length === 3)) {
-    avatarContent = <span className={avatarTextClassNames}>{initials.toUpperCase()}</span>;
-  } else {
-    avatarContent = icon;
+      const imageComponent = Avatar.generateImage(image, variant, alt);
+      this.setState({ imageComponent });
+    }
   }
 
-  return (
-    <div {...attributes} aria-label={ariaLabel} className={avatarClassNames}>
-      {avatarContent}
-    </div>
-  );
-};
+  render() {
+    const { alt,
+      ariaLabel,
+      initials,
+      variant,
+      image,
+      ...customProps } = this.props;
+
+    const attributes = Object.assign({}, customProps);
+
+    const avatarIconClassNames = cx([
+      'avatar-icon',
+      variant,
+    ]);
+
+    const avatarClassNames = cx([
+      'avatar',
+      attributes.className,
+    ]);
+
+    const avatarTextClassNames = cx([
+      { 'avatar-text-small': initials && initials.length === 3 },
+      { 'avatar-text-large': initials && initials.length === 2 },
+    ]);
+
+    const icon = <span className={avatarIconClassNames} />;
```

Comment:
This needs removed, along with `avatarIconClassNames` as you moved where this is used.